miden-lib/asm/kernels/transaction/main.masm

```diff
@@ -95,6 +95,10 @@ proc.main.1
         exec.note::prepare_note
         # => [NOTE_SCRIPT_HASH, NOTE_ARGS]
 
+        # store the NOTE_SCRIPT_HASH to the memory to be able to call it dynamically
+        exec.memory::store_dyn_proc_root
+        # => [note_script_hash_addr, NOTE_ARGS]
```

Comment:
Similar comment to the one above: seems like we may be doing double work here by first getting script hash from memory and then storing it there again.